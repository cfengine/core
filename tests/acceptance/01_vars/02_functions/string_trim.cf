body common control
{
      inputs => { "../../default.cf.sub" };
      bundlesequence  => { default("$(this.promise_filename)") };
      version => "1.0";
}

#######################################################

bundle agent test
{
  meta:
      "test_soft_fail" string => "windows",
        meta => { "ENT-10254" };

  vars:
      "abcd1" string => string_trim("abcd");
      "abcd2" string => string_trim(" abcd");
      "abcd3" string => string_trim("abcd ");
      "abcd4" string => string_trim("         abcd               ");
      "abcd5" string => string_trim("abcd

      ");
      "abcd6" string => string_trim("
      
      abcd");
      "abcd7" string => string_trim("   
      
      abcd
      
      ");
      "abcd_abcd" string => string_trim("  abcd    abcd      ");

      # empty cases
      "empty1" string => string_trim("");
      "empty2" string => string_trim("     ");
      "empty3" string => string_trim("   
      
      
      ");
}

#######################################################

bundle agent check
{
  methods:
      "" usebundle => dcs_check_state(test,
                                      "$(this.promise_filename).expected.json",
                                       $(this.promise_filename));
}
