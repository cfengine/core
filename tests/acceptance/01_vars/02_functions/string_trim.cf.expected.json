{
  "abcd1": "abcd",
  "abcd2": "abcd",
  "abcd3": "abcd",
  "abcd4": "abcd",
  "abcd5": "abcd",
  "abcd6": "abcd",
  "abcd7": "abcd",
  "abcd_abcd": "abcd    abcd",
  "empty1": "",
  "empty2": "",
  "empty3": ""
}
